### PR TITLE
VPN-2724 - 🚧 [Prototype] Use the Glean Rust SDK to collect telemetry 🚧

### PR DESCRIPTION
**Pull request by brizental**
I built this prototype to see if it would be possible and manageable to use the Glean Rust based SDKs instead of Glean.js on the VPN client. See more about the motivation for this on: [Proposal: Use the Glean Rust-based SDKs on the Mozilla VPN client application](https://docs.google.com/document/d/1M5AZXe2b5k9_Eig15OBuBc7vhUwRA1FWdBs4VcL8LAE/edit?usp=sharing).

What is implemented here:

- Use the Glean Rust SDK + wrapper to collect metrics from the C++ layer of the VPN client;
- Use the Glean iOS SDK to initialize Glean on iOS and levarage the iOS specific goodies that Glean provides.

What is not implemented here:

- Use the Glean Rust SDK to collect metrics from the QML layer of the VPN client;
- Link the Glean Android SDK to the same Glean library used on C++;
- Implement all metrics currently recorded by the VPN using the Rust SDK;
- Feature flag to switch between Rust / JS Glean SDKs;
- Proper logging. Only the iOS logs are currently working;
- Nice and clean implementation + Documentation + Testing;
- (Probably even more stuff.)

### CODE REVIEW

File: scripts/macos/apple_compile.sh [hunk 1]
```diff
@@ -99,6 +99,37 @@ if [[ "$OS" == "ios" ]]; then
...
 fi
 
+if [[ "$OS" == "ios" ]]; then
+  printn Y "Glean stuff... \n"
+  
+  # Version MUST be kept in sync with Glean's version. We should probably automate the syncing.
+  cargo install uniffi_bindgen@0.19.6
+  mkdir -p ios/Glean
+  # Generate the FFI stuff for the Glean iOS SDK
+  uniffi-bindgen generate -l swift -o "ios/Glean" "3rdparty/glean/glean-core/src/glean.udl"
```
Comment (badboy) [inline scripts/macos/apple_compile.sh:109]: If you vendor glean anyway, you should be able to use the `embedded-uniffi-bindgen` crate and thus not need to sync the uniffi-bindgen version:

```suggestion
cargo run --manifest-path 3rdparty/glean/tools/embedded-uniffi-bindgen/Cargo.toml -- generate -l swift -o "ios/Glean" "3rdparty/glean/glean-core/src/glean.udl"
```

File: scripts/macos/apple_compile.sh [hunk 2]
```diff
@@ -99,6 +99,37 @@ if [[ "$OS" == "ios" ]]; then
...
+  # Rename the generated file otherwise it will prevent the Glean.swift file from being copied in the next step
+  cp ios/Glean/glean.swift ios/Glean/GeneratedGlean.swift
+  rm ios/Glean/glean.swift
```
Comment (badboy) [inline scripts/macos/apple_compile.sh:112]:
```suggestion
  mv ios/Glean/glean.swift ios/Glean/GeneratedGlean.swift
```

File: scripts/macos/apple_compile.sh (hunk 1 above)
Comment (brizental) [reply scripts/macos/apple_compile.sh:109]: Nice!

**Comment by brizental** [comment]
Closing this because it's stale and everything that is related to it has been / will be merged in other PRs (see tickets related to https://mozilla-hub.atlassian.net/browse/VPN-3054). The prototype has served it's purpose.

**Pull request closed without merging**